lib/asset/AssetModulesPlugin.js

```diff
@@ -49,7 +74,10 @@ class AssetModulesPlugin {
 						chunk,
 						compareModulesByIdentifier
 					)) {
-						if (module.getSourceTypes().has("asset")) {
+						if (
+							module.getSourceTypes().has("asset") &&
+							!module.buildMeta.dataUrlAsset
```

Comment:
this is a little hack. I think it's better no not have `source type` `asset` at first place. Note source type is also used at other places e. g. in stats or for splitChunks.

For this you have to avoid `Generator.byType` because it can only return a fixed list of source types. Instead merge AssetJavascriptGenerator into AssetGenerator, select behavior based on `type` argument of the functions (read Generator.byType for how it works). When getting source types return `javascript` or `javascript and asset` based on content size and options.